### PR TITLE
TINY-11395: Fix inability to type '{' character on German keyboard layouts

### PR DESCRIPTION
**Pull request by ArvinJ-H**
Related Ticket: 
TINY-11395

Description of Changes:
- Changed the keyboard shortcut to a custom keyboard handler

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Comment by TheSpyder** [comment]
@spocke I think we wanted a more generic fix not a patch to the one we know is a problem?

File: modules/tinymce/src/core/main/ts/api/NotificationManager.ts [hunk 1]
```diff
@@ -155,10 +155,14 @@ const NotificationManager = (editor: Editor): NotificationManager => {
       });
     });
 
-    editor.addShortcut('alt+F12', 'Focus to notification', () =>
-      getTopNotification()
-        .map((notificationApi) => SugarElement.fromDom(notificationApi.getEl()))
-        .each((elm) => Focus.focus(elm)));
+    editor.on('keydown', (e) => {
+      if ((e.altKey || e.metaKey) && e.key.toLowerCase() === 'f12') {
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/api/NotificationManager.ts:159]: Not sure we should allow both meta and alt on both Mac OS and Windows/Linux?
Are we sure meta is even right on any platform the old key binding seems to be alt only?

**Review by spocke — CHANGES_REQUESTED** [review]
Is there no tests for this, then we should add that. Since agar doesn't yet support this new api we can use `dom.dispatchEvent(new KeyboardEvent())` in the meantime.

**Comment by spocke** [comment]
> @spocke I think we wanted a more generic fix not a patch to the one we know is a problem?

I think we can make a general fix later it would be a bit more work and testing thst this and less critical.

File: modules/tinymce/src/core/main/ts/api/NotificationManager.ts (hunk 1 above)
Comment (TheSpyder) [reply modules/tinymce/src/core/main/ts/api/NotificationManager.ts:159]: `meta` is `command` on macOS. So no, we should not allow meta.

Comment (ArvinJ-H) [reply modules/tinymce/src/core/main/ts/api/NotificationManager.ts:159]: yeah the `meta` key was a mistake, sorry about that

File: modules/tinymce/src/core/main/ts/api/NotificationManager.ts [hunk 2]
```diff
@@ -155,10 +155,14 @@ const NotificationManager = (editor: Editor): NotificationManager => {
       });
     });
 
-    editor.addShortcut('alt+F12', 'Focus to notification', () =>
-      getTopNotification()
-        .map((notificationApi) => SugarElement.fromDom(notificationApi.getEl()))
-        .each((elm) => Focus.focus(elm)));
+    editor.on('keydown', (e) => {
+      if ((e.altKey) && e.key.toLowerCase() === 'f12') {
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/api/NotificationManager.ts:159]:
```suggestion
      if (e.altKey && e.key.toLowerCase() === 'f12') {
```

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/api/NotificationManager.ts (hunk 2 above)
Comment (spocke) [reply modules/tinymce/src/core/main/ts/api/NotificationManager.ts:159]: This change really needs a test if it's pressed for time log a follow up to add a test.

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/api/NotificationManager.ts (hunk 2 above)
Comment (ArvinJ-H) [reply modules/tinymce/src/core/main/ts/api/NotificationManager.ts]: @spocke There is a test case for focusing to notification, it was still using `key code` I updated to use keyboard keypress

**Review by MitchC1999 — APPROVED** [review]

**Review by kimwoodfield — APPROVED** [review]

**Review by shanmen-tiny — APPROVED** [review]

**Comment by ArvinJ-H** [comment]
> > @spocke I think we wanted a more generic fix not a patch to the one we know is a problem?
> 
> I think we can make a general fix later it would be a bit more work and testing that this and less critical.

@spocke @TheSpyder Created a ticket for this. https://ephocks.atlassian.net/browse/TINY-11429

File: .changes/unreleased/tinymce-TINY-11395-2024-10-17.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Fixed
+body: Fixed inability to type '{' character on German keyboard layouts.
```
Comment (TheSpyder) [inline .changes/unreleased/tinymce-TINY-11395-2024-10-17.yaml:3]: I missed this, we don't need `fixed` in the body for fixes

File: modules/tinymce/src/core/main/ts/api/NotificationManager.ts [hunk 4]
```diff
@@ -155,10 +155,16 @@ const NotificationManager = (editor: Editor): NotificationManager => {
       });
     });
 
-    editor.addShortcut('alt+F12', 'Focus to notification', () =>
-      getTopNotification()
-        .map((notificationApi) => SugarElement.fromDom(notificationApi.getEl()))
-        .each((elm) => Focus.focus(elm)));
+    editor.on('keydown', (e) => {
+      // TODO: Remove this once we remove the use of keycodes
```
[the remark this answers is not in the document]
Comment (ArvinJ-H) [reply modules/tinymce/src/core/main/ts/api/NotificationManager.ts:159]: https://github.com/tinymce/tinymce/pull/9939

**Pull request merged**